doc/index.rst

```diff
@@ -40,6 +40,7 @@ Sections
    api/api.rst
    samples/samples.rst
    reference/kconfig/index.rst
+   tools/board-support-tools.rst
```

Comment:
That's why I had it originally in the boards.rst page.  

Do you want the files in the docs/ area, but keep the links to them at the bottom the boards.rst page as I had, or don't have these board tools docs show up in the index at all (just as references from docs that want to point them out)?  I can do either.